src/main/java/Deadline.java

```diff
@@ -0,0 +1,26 @@
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+
+public class Deadline extends Task implements Serializable {
+
+    protected String by;
+    LocalDate date;
+
+    public Deadline(String description, String by) {
+        super(description);
+        try {
+            this.date = LocalDate.parse(by);
```

Comment:
Perhaps `this` is unnecessary? The basic coding standard doesn't require this, but is suggested by the advanced style guide.
```suggestion
            date = LocalDate.parse(by);
```
I noticed this in several other places too.